include/villas/nodes/c37_118/parser.hpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file
+ * @author Philipp Jungkamp <philipp.jungkamp@rwth-aachen.de>
+ * @copyright 2014-2022, Institute for Automation of Complex Power Systems, EONERC
+ * @license Apache 2.0
+ *********************************************************************************/
+
+#pragma once
+
+#include <optional>
+#include <cstring>
+#include <villas/nodes/c37_118/types.hpp>
+
+namespace villas {
+namespace node {
+namespace c37_118 {
+namespace parser {
+
+using Result = std::intptr_t;
+
+enum class Status : Result {
+	Ok = 0,
+	MissingBytes,
+	MissingConfig,
+	InvalidValue,
+	InvalidChecksum,
+	InvalidSlice,
+	Other,
+};
+
+template <typename T = Result>
+Result ok(T t = 0);
+Result error(Status);
+Status status(Result);
+bool is_ok(Result);
+
+// context used to assemble and disassemble data frames with the information contained in config structs
```

Comment:
Please make comments a sentence: start with capital letter and end with period.
```suggestion
// Context used to assemble and disassemble data frames with the information contained in config structs.
```